File: C++/Matriz/diagonal negativa.cpp
#include<iostream>

using namespace std;
int main(){

    int i,j,n;

    cout << "Qual a ordem da matriz? ";
    cin >> n;
            int vet[n][n], negativo=0;

            for (i=0; i<n; i++){
                for (j=0; j<n; j++){
                    cout << "Elemento [" << i << "," << j << "]:";
                    cin >> vet[i][j];

                }
            }
            cout << "\nDIAGONAL PRINCIPAL:"<<endl;
            for (i=0; i<n; i++){
                for (j=0; j<n; j++){
                if (i==j){
                        cout << vet[i][j] << " ";
                    }
                if (vet[i][j] < 0){
                    negativo = negativo + 1;
                }
                }
            }
            cout << "\nQUANTIDADE DE NEGATIVOS = " << negativo;



}

File: Phyton/Matriz/diagonalNegativa.py
n = int(input("Qual a ordem da matriz ?"))

mat = [[0 for x in range(n)]for x in range(n)]

for i in range(n):
    for j in range(n):
        mat[i][j] = int(input(f"Elemento [{i},{j}]:"))

print("Diagonal Principal: ")
negativosMatriz = 0
for i in range(n):
    for j in range(n):
        if i == j:
            print(mat[i][j],end=" ")
        if mat[i][j] < 0:
            negativosMatriz += 1
print()
print(f"QUANTIDADE DE NEGATIVOS = {negativosMatriz}")